pkg/server/plugin/upstreamauthority/gcpca/gcpca.go

```diff
@@ -0,0 +1,284 @@
+)
+
+// BuiltIn constructs a catalog Plugin using a new instance of this plugin.
+func BuiltIn() catalog.Plugin {
+	return builtin(New())
+}
+
+func builtin(p *Plugin) catalog.Plugin {
+	return catalog.MakePlugin(pluginName, upstreamauthority.PluginServer(p))
+}
+
+type Config struct {
+	Project  string `hcl:"gcp_project_name"`
+	Location string `hcl:"gcp_region_name"`
+	CertName string `hcl:"gcp_cert_name"`
+}
+
+type Plugin struct {
+	// gRPC requires embedding either the "Unimplemented" or "Unsafe" stub as
+	// a way of opting in or out of forward build compatibility.
+	upstreamauthority.UnimplementedUpstreamAuthorityServer
+
+	// mu is a mutex that protects the configuration. Plugins may at some point
+	// need to support hot-reloading of configuration (by receiving another
+	// call to Configure). So we need to prevent the configuration from
+	// being used concurrently and make sure it is updated atomically.
+	mu sync.Mutex
+	c  *Config
+
+	log hclog.Logger
+}
+
+// These are compile time assertions that the plugin matches the interfaces the
+// catalog requires to provide the plugin with a logger and host service
+// broker as well as the UpstreamAuthority itself.
+var _ catalog.NeedsLogger = (*Plugin)(nil)
+var _ catalog.NeedsLogger = (*Plugin)(nil)
+var _ upstreamauthority.UpstreamAuthorityServer = (*Plugin)(nil)
+
+func New() *Plugin {
+	return &Plugin{}
+}
+
+// SetLogger will be called by the catalog system to provide the plugin with
+// a logger when it is loaded. The logger is wired up to the SPIRE core
+// logger
+func (p *Plugin) SetLogger(log hclog.Logger) {
+	// TODO: store the logger for later use. If the plugin does not need to
+	// log, this method can be removed.
+	p.log = log
+}
+
+func getClient(ctx context.Context) (*pcaapi.CertificateAuthorityClient, error) {
+	creds, err := google.FindDefaultCredentials(ctx)
+
+	if err != nil {
+		return nil, err
+	}
+
+	pcaClient, err := pcaapi.NewCertificateAuthorityClient(ctx, option.WithCredentials(creds))
+
+	if err != nil {
+		return nil, err
+	}
+
+	return pcaClient, nil
+}
+
+func pemToX509(pemStr string) (*x509.Certificate, error) {
+	block, rest := pem.Decode([]byte(pemStr))
+	if len(rest) > 0 {
+		return nil, errors.New("Unexpected trailer in certificate of size " + fmt.Sprint(len(rest)))
+	}
+
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return cert, nil
+}
+
+func (p *Plugin) mintX509CA_(log hclog.Logger, ctx context.Context, csr []byte, preferredTtl int32) (*upstreamauthority.MintX509CAResponse, error) {
+	csrParsed, err := x509.ParseCertificateRequest(csr)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse CSR: %v", err)
+	}
+
+	validity := time.Second * time.Duration(preferredTtl)
+	parentPath := "projects/" + p.c.Project + "/locations/" + p.c.Location + "/certificateAuthorities/" + p.c.CertName
+
+	log.Info("Minting X509 using certificate at " + parentPath + "; ttl " + validity.String())
```

Comment:
nitpick: I'd suggest making this more structured
```suggestion
	log.Info("Minting X509 intermediate CA ", "ca-certificate", parentPath, "ttl ", validity)
```